### PR TITLE
feat: add smb driver (close #1746)

### LINKED ISSUE #1746: 希望v3能挂载电脑共享smb
### Please make sure of the following things

- [ ] I have read the [documentation](https://alist.nn.ci).
- [ ] I'm sure there are no duplicate issues or discussions.
- [ ] I'm sure this feature is not implemented.
- [ ] I'm sure it's a reasonable and popular requirement.

### Description of the feature / 需求描述

希望v3能挂载电脑共享smb

### Suggested solution / 实现思路

_No response_

### Additional context / 附件

_No response_

### PR DESCRIPTION
**Pull request by BoYanZh**

### CODE REVIEW

File: drivers/smb/driver.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+}
+
+//func (d *SMB) Get(ctx context.Context, path string) (model.Obj, error) {
+//	// this is optional
+//	return nil, errs.NotImplement
+//}
+
+func (d *SMB) Link(ctx context.Context, file model.Obj, args model.LinkArgs) (*model.Link, error) {
+	fullPath := file.GetPath()
+	return &model.Link{FilePath: &fullPath}, nil
```
Comment (Xhofe) [inline drivers/smb/driver.go:73]: Is it correct to return the local file path here?

Comment (BoYanZh) [reply drivers/smb/driver.go:73]: Yes, I have tried to return the path with the prefix `.` and it gives a 500 error, which takes me some time to find out.

Comment (Xhofe) [reply drivers/smb/driver.go:73]: But wouldn't that just return the native file only? 

it is possible that smb on another machine.

Comment (BoYanZh) [reply drivers/smb/driver.go:73]: My bad, I think I do not understand the meaning of the function, trying to fix it now.

Comment (BoYanZh) [reply drivers/smb/driver.go]: I think it works now for smb on another machine.

Comment (Xhofe) [reply drivers/smb/driver.go]: Ok. This is function return one of three:
- path of native file
- download url of web file
- reader closer

For this, It maybe like:
```go
f,_ := d.fs.Open(converted_path)
return &Link{
    Data: f
}
```
In a rough sense

**Comment by Xhofe** [comment]
Looks super - thank you. I wounld merge it after ci finished.

**Comment by SheltonZhu** [comment]
今天试了下，好用！正好有这个需求。
就是今天rootpath配置错了的时候，把主程序弄奔溃了

**Pull request merged**